src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -858,6 +958,24 @@ void AggregateEntityResourceInternal(long entityId, long resourceEntityId)
                 })
                 .ToList();
 
+            var entityNonFungibleVaultHistoryToAdd = nonFungibleVaultChanges
+                .Select(e =>
+                {
+                    AggregateEntityResource(e.ReferencedVault, e.ReferencedResource, e.StateVersion, x => x.NonFungibleResourceEntityIds, x => x.NonFungibleResourceLastUpdateStateVersions);
```

Comment:
It took my a lot of time to understand what's going on in here. 

As you already mentioned `PostgresLedgerExtenderService` deserves some care/refactoring.  Just wanted to raise it here so we can schedule/agree on some actions to perform.

We're calling local function which is placed in the middle of upper function (in one of `case` scenarios), which then calls yet another inner local function which manipulates some local variable from 3 scopes up. Overall complexity of entire logic here is very hard to understand and I bet it's gonna be almost impossible to write/change any code here for me.